### PR TITLE
Fix chunk <-> block confusion providing comments to explain what is what

### PR DESCRIPTION
**Pull request by zakarumych**
This change is [Reviewable](https://reviewable.io/reviews/gfx-rs/gfx-mem/6)

### CODE REVIEW

File: src/chunked.rs [hunk 1]
```diff
@@ -10,19 +10,27 @@ use gfx_hal::memory::Requirements;
 use {alignment_shift, MemoryAllocator, MemoryError, MemorySubAllocator};
 use block::{Block, RawBlock};
 
+/// Chunks is super-allocator block
```
Comment (kvark) [inline src/chunked.rs]: "Chunks is"

File: src/chunked.rs [hunk 2]
```diff
@@ -10,19 +10,27 @@ use gfx_hal::memory::Requirements;
...
+/// That is devided into smaller blocks
```
Comment (kvark) [inline src/chunked.rs]: as this starts with capital letter, we need to end the previous sentence with a period.

File: src/chunked.rs [hunk 3]
```diff
@@ -111,36 +133,49 @@ where
...
-            self.blocks[block_index].memory(),
+            self.chunks[chunk_index].memory(),
             block_memory
         ));
-        let chunk_index = offset / self.chunk_size;
+
+        // Calculate block index inside chunk
+        let block_index = (offset - self.chunks[chunk_index].range().start) / self.block_size;
+
+        // Push into free blocks list
```
Comment (kvark) [inline src/chunked.rs]: not sure if omitting all the prepositions is good here, spreading "a"/"the" before used terms may make the comments more readable and add a bit of semantics

File: src/chunked.rs (hunk 1 above)
Comment (mjadczak) [reply src/chunked.rs]: > Chunks are super-allocator blocks, which are then divided into smaller 'blocks'

File: src/chunked.rs [hunk 4]
```diff
@@ -32,19 +40,22 @@ impl<T> ChunkedNode<T> {
...
     }
 
     fn is_used(&self) -> bool {
+        // All blocks are free
         self.count() != self.free.len()
     }
 
     fn count(&self) -> usize {
-        self.blocks.len() * self.blocks_per_chunk()
+        // Blocks count is chunk count multiplyed by blocks per chunk
```
Comment (mjadczak) [inline src/chunked.rs]: multiplyed -> multiplied

File: src/chunked.rs [hunk 5]
```diff
@@ -32,19 +40,22 @@ impl<T> ChunkedNode<T> {
...
+        self.chunks.len() * self.blocks_per_chunk()
     }
 
     fn blocks_per_chunk(&self) -> usize {
+        // How much blocks there in chunk
```
Comment (mjadczak) [inline src/chunked.rs]: This comment is probably redundant given the function name, but should be
> How many blocks there are in a chunk

File: src/chunked.rs [hunk 6]
```diff
@@ -62,19 +73,26 @@ impl<T> ChunkedNode<T> {
         let reqs = Requirements {
             type_mask: 1 << self.id.0,
             size: self.chunk_size,
-            alignment: self.chunk_size,
+            alignment: self.block_size,
         };
-        let block = owner.alloc(device, request, reqs)?;
-        assert_eq!(0, alignment_shift(reqs.alignment, block.range().start));
-        assert!(block.size() >= self.chunk_size);
+        // Get new chunk
```
Comment (mjadczak) [inline src/chunked.rs]: > Get a new chunk

File: src/chunked.rs [hunk 7]
```diff
@@ -62,19 +73,26 @@ impl<T> ChunkedNode<T> {
...
+        let chunk = owner.alloc(device, request, reqs)?;
+        assert_eq!(0, alignment_shift(reqs.alignment, chunk.range().start));
+        assert!(chunk.size() >= self.chunk_size);
 
         let blocks_per_chunk = self.blocks_per_chunk();
-        let block_index = self.blocks.len();
+
+        // It takes next index
```
Comment (mjadczak) [inline src/chunked.rs]: > `len()` will return the next index to use

File: src/chunked.rs [hunk 8]
```diff
@@ -62,19 +73,26 @@ impl<T> ChunkedNode<T> {
...
+        let chunk_index = self.chunks.len();
+
+        // Fill free list with new blocks
```
Comment (mjadczak) [inline src/chunked.rs]: > Fill the free list with new blocks

File: src/chunked.rs [hunk 9]
```diff
@@ -62,19 +73,26 @@ impl<T> ChunkedNode<T> {
...
         self.free.extend((0..blocks_per_chunk).map(|i| FreeBlock {
-            block_index,
-            chunk_index: i as u64,
+            chunk_index,
+            block_index: i as u64,
         }));
-        self.blocks.push(block);
+
+        // Put new chunk in list
```
Comment (mjadczak) [inline src/chunked.rs]: > Place the new chunk in the list

File: src/chunked.rs [hunk 10]
```diff
@@ -84,13 +102,17 @@ impl<T> ChunkedNode<T> {
         M: Debug + Any,
         T: Block<Memory = M>,
     {
+        // Find free block
```
Comment (mjadczak) [inline src/chunked.rs]: > Find a free block

File: src/chunked.rs [hunk 11]
```diff
@@ -84,13 +102,17 @@ impl<T> ChunkedNode<T> {
...
+            // plus chunk memory offset
+            let offset = free_block.block_index * self.block_size + self.chunks[free_block.chunk_index].range().start;
             let block = RawBlock::new(
-                self.blocks[free_block.block_index].memory(),
-                offset..self.chunk_size + offset,
+                self.chunks[free_block.chunk_index].memory(),
+                offset..self.block_size + offset,
             );
-            ChunkedBlock(block, free_block.block_index)
+            // Remember what chunk it came from
```
Comment (mjadczak) [inline src/chunked.rs]: > Remember what chunk the block came from

File: src/chunked.rs [hunk 12]
```diff
@@ -111,36 +133,49 @@ where
...
             return Err(MemoryError::NoCompatibleMemoryType);
         }
-        Ok(match self.alloc_no_grow() {
-            Some(block) => {
-                assert!(block.size() >= reqs.size);
-                assert_eq!(block.range().start & (reqs.alignment - 1), 0);
-                block
-            }
+
+        // Try to allocate block
```
Comment (mjadczak) [inline src/chunked.rs]: > Try to allocate a block

File: src/chunked.rs [hunk 13]
```diff
@@ -111,36 +133,49 @@ where
...
         let block_memory: *const B::Memory = block.memory();
-        let block_index = unsafe {
+
+        // Dispose block retreiving chunk index
+        let chunk_index = unsafe {
             block.0.dispose();
             block.1
         };
+
+        // Confirm chunk index
```
Comment (mjadczak) [inline src/chunked.rs]: > Confirm the chunk index

File: src/chunked.rs [hunk 14]
```diff
@@ -111,36 +133,49 @@ where
...
         assert!(::std::ptr::eq(
-            self.blocks[block_index].memory(),
+            self.chunks[chunk_index].memory(),
             block_memory
         ));
-        let chunk_index = offset / self.chunk_size;
+
+        // Calculate block index inside chunk
```
Comment (mjadczak) [inline src/chunked.rs]: > Calculate the block index inside the chunk

File: src/chunked.rs (hunk 3 above)
Comment (mjadczak) [inline src/chunked.rs]: > Push the block back into the 'free blocks' list

**Review by mjadczak — CHANGES_REQUESTED** [review]
I agree with @kvark that the clarity of the wording could be a little improved—I went through and suggested the wording to use in each case.

**Review by kvark — APPROVED** [review]
nice, thanks!
bors r+

**Pull request merged**